### DISCUSSION TITLE
Metadata v3 spec

### DISCUSSION THREAD

**Discussion by joshstevens19** [aave/lens-api-examples#101] [Ideas]
gm All! Based on ecosystem feedback we are going to be beginning the process for creating the next version of the metadata standard: Metadata V3! Just like V2, we hope to have the proposal drafted with developers from around the community, including all of you. We are hoping to target release for late nov/early december.

We have heard a lot of requests for support for feature flags, licenses, music support features and more!

Please drop any idea you have on metadata V3 and we can all discuss. With this we can then create a final draft of what is in it and what is not!

**Comment by haithamelmengad** [comment]
gm! This is a great initiative and will certainly lead to the most composable metadata standard. Here's one high level suggestion:

Two overarching fields, "protocol" and "body": protocol identifies the smart contracts that this NFT was minted by and is interacting with, by default this field is "Lens", and body has all the meta-data fields and contents. This allows interoperability between Lens Protocol NFTs and other NFTs minted by external smart contracts. These NFTs may have fields that are required by their domain application for custom logic, and are unnecessary to include in the general Lens protocol metadata standard.

I will also follow up with a proposal for music/media NFT meta-data fields.

**Reply by joshstevens19** [reply]
Interesting, this is too big of a breaking change to roll out with the new spec; it would also mean it would be out of spec with ERC721 and not show on the likes of opensea and others. I like what you are getting on, though. Are you thinking of being able to construct a blockchain transaction event? like "I just brought this NFT"..?

**Comment by ShravanSunder** [comment]
As a Reference:
-  this is the current lens v2 metadata: https://docs.lens.xyz/docs/metadata-standards

```ts
interface PublicationMetadataMedia {
  item: Url;
  /**
   * This is the mime type of media
   */
  type?: MimeType | null;
  
  /**
   * The alt tags for accessibility
   */
  altTag?: string | null;

  /**
   * The cover for any video or audio you attached
   */
  cover?: string | null;
}

enum PublicationMetadataVersions {
  one = '1.0.0',
  // please use metadata v2 when doing anything! v1 is supported but discontinued.
  two = '2.0.0'
}

enum PublicationMetadataDisplayType {
  number = 'number',
  string = 'string',
  date = 'date',
}

interface PublicationMetadataAttribute {
  displayType?: MetadataDisplayType | undefined | null;
  traitType?: string | undefined | null;
  value: string;
}

enum PublicationContentWarning {
  NSFW = 'NSFW',
  SENSITIVE = 'SENSITIVE',
  SPOILER = 'SPOILER',
}

enum PublicationMainFocus {
  VIDEO = 'VIDEO',
  IMAGE = 'IMAGE',
  ARTICLE = 'ARTICLE',
  TEXT_ONLY = 'TEXT_ONLY',
  AUDIO = 'AUDIO',
  LINK = 'LINK',
  EMBED = 'EMBED',
}

interface Metadata {
  /**
   * The metadata version.
   */
  version: PublicationMetadataVersions;

   /**
   * The metadata lens_id can be anything but if your uploading to ipfs
   * you will want it to be random.. using uuid could be an option!
   */
  metadata_id: string;

  /**
   * A human-readable description of the item.
   */
  description?: Markdown | undefined | null;

  /**
   * The content of a publication. If this is blank `media` must be defined or its out of spec.
   */
  content?: Markdown | undefined | null;
  
  /**
   * IOS 639-1 language code aka en or it and ISO 3166-1 alpha-2 region code aka US or IT aka en-US or it-IT
   * Full spec > https://tools.ietf.org/search/bcp47
   */
  locale: Locale;

  /**
   * Ability to tag your publication
   */
  tags?: string[] | undefined | null;

  /**
   * Ability to add a content warning
   */
  contentWarning?: PublicationContentWarning | undefined | null;

  /**
   * Main content focus that for this publication
   */
  mainContentFocus: PublicationMainFocus;

  /**
   * This is the URL that will appear below the asset's image on OpenSea and others etc
   * and will allow users to leave OpenSea and view the item on the site.
   */
  external_url?: Url | undefined | null;

  /**
   * Name of the item.
   */
  name: string;

  /**
   * These are the attributes for the item, which will show up on the OpenSea and others NFT trading websites on the 
  item.
   */
  attributes?: PublicationMetadataAttribute[] | undefined | null;

  /**
   * legacy to support OpenSea will store any NFT image here.
   */
  image?: Url | undefined | null;

  /**
   * This is the mime type of image. This is used if you uploading more advanced cover images
   * as sometimes IPFS does not emit the content header so this solves the pr
   */
  imageMimeType?: MimeType | undefined | null;

  /**
   * This is lens supported attached media items to the publication
   */
  media?: PublicationMetadataMedia[] | undefined | null;

  /**
   * In spec for OpenSea and other providers - also used when using EMBED main publication focus
   * A URL to a multi-media attachment for the item. The file extensions GLTF, GLB, WEBM, MP4, M4V, OGV,
   * and OGG are supported, along with the audio-only extensions MP3, WAV, and OGA.
   * Animation_url also supports HTML pages, allowing you to build rich experiences and interactive NFTs using JavaScript canvas,
   * WebGL, and more. Scripts and relative paths within the HTML page are now supported. However, access to browser extensions is not supported.

   */
  animation_url?: Url | undefined | null;

  /**
   * This is the appId the content belongs to
   */
  appId?: AppId | undefined | null;
}
```

**Comment by ShravanSunder** [comment]
I want to have the ability to link or embed full length articles with a summary in `content`.  And do it in a way that would be standard with different lens apps displaying this.  

## Thoughts on how to do this current v2.  

If we wanted to have html or json context payload, was the idea to sent `content = null` and `PublicationMetadataMedia.type = text\html` with an `PublicationMetadataMedia.item`.  Alternatively i was thinking, i could have content that is markdown as a summary and then the actual article with the `PublicationMetadataMedia.item`

If not how do you see us loading content in a way that can be server side rendered.  I know Markdown is an acceptable for `content`.  However, markdown content could render faster for the user if its renderd on the server with SSR.

> i might be missing on a easier way to do this in v2

## Conclusion

I think having a more defined way to have a summary in lens and a way to either link or embed full length articles would be great.

**Reply by cesarenaldi** [reply]
A way to achieve this in v2 is to:
- set `mainContentFocus` to `LINK`
- write your excerpt in the `content` field
- add the URL of the full article in the `content`. It could be also be a nice `[Read more](URL)` MD link.

Although medias could in theory be used for HTML content, I would personally refrain from using HTML as data exchange format. Many FE will be aware of the risk associated with rendering HTML within their applications and they are likely to sandbox it or omit it. 

The reason `content` is Markdown is because syntactically is virtually impossible to write wrong markdown (semantically is a different story but that goes beyond what a JSON specification can enforce). It makes for a great text exchange format and has some semantic primitives that can be used to inform FEs about the formatting/rendering needs.

Maybe worth exploring if there are other data exchange formats with similar (a more) capabilities to Markdown and entertain the idea of using those. What do you think?

**Reply by cesarenaldi** [reply]
Also embed are supported in v2, although the "full length" article won't be part of the metadata JSON but is an external resource.

Have  look at [`mainContentFocus=EMBED`](https://docs.lens.xyz/docs/metadata-standards#embed). You can link the embeddable content URL via the `animated_url` property. You can then still use the `content` to have an excerpt/summary of the full article.

**Reply by ShravanSunder** [reply]
thanks i'll look at this as i try things out. I appreciate it

**Reply by joshstevens19** [reply]
already in v2 feature set

**Comment by EthWarrior** [comment]
I propose to add also "license" details into the metadata. Lens Protocol enables ownership of your social graph and public content directly to your audience. Following this idea, would be suitable if the profile owners (or applications for them) could submit the license type of their content. Hence someone who posts could submit CCO or reserve copyright. Would be also good to have a standard list of commonly used rights for consistency.

**Reply by oneski** [reply]
strongly support adding a license field, would propose initial support be an enum of common licenses

Initial list would include: CC 4.0 licenses (+ cc0), a16z's CBE licenses.

Would love to hear if there are any other specific ones we should include in initial enum for v3.

CC: https://creativecommons.org/about/cclicenses/
CBE: https://a16zcrypto.com/introducing-nft-licenses/

**Reply by joshstevens19** [reply]
yes license will be part of metadata v3 for sure

**Comment by sweetmantech** [comment]
I propose we support full music metadata for audio metadata. Will help encourage adoption of music nfts in the Lens ecosystem. 💿📻

- Metadata Schema: https://gist.github.com/bretth18/df8358c840fa94946ec212f753e290dd
- Live Deployment on Polygon Mainnet in `MetadataRenderer` Architecture for 100% onchain implementation on Polygon (we don't need 100% onchain but definitely feasible on Polygon): [0xccff064c15e36be681ee2807fce8fe38176e8baa](https://polygonscan.com/address/0xccff064c15e36be681ee2807fce8fe38176e8baa#code)

Some example fields:
- `losslessAudio`
- `bpm`
- `duration`
- `license`
- `credits`
- `artist`
- and many more. view full schema (above) for full list of supported metadata.

**Reply by cdsiren** [reply]
[decent](https://github.com/decentxyz/DecentSDK-protocol) team can support on implementation, etc. if group elects to adopt

**Reply by cdsiren** [reply]
> I propose we support full music metadata for audio metadata. Will help encourage adoption of music nfts in the Lens ecosystem. 💿📻
> 
> * Metadata Schema: https://gist.github.com/bretth18/df8358c840fa94946ec212f753e290dd
> * Live Deployment on Polygon Mainnet in `MetadataRenderer` Architecture for 100% onchain implementation on Polygon (we don't need 100% onchain but definitely feasible on Polygon): [0xccff064c15e36be681ee2807fce8fe38176e8baa](https://polygonscan.com/address/0xccff064c15e36be681ee2807fce8fe38176e8baa#code)
> 
> Some example fields:
> 
> * `losslessAudio`
> * `bpm`
> * `duration`
> * `license`
> * `credits`
> * `artist`
> * and many more. view full schema (above) for full list of supported metadata.

Another point I came to realize w. the Ooh La La team:
Right now whoever posts a song on Lens (e.g., [Christina](https://www.oohlala.xyz/artist/8c7f4751-935e-4c15-a5db-559e917ae8c2)) becomes the "artist" according to indexers.  We can / should enforce the `artist name` in this metadata standard as a required field so that it remains constant even as - particularly CC0 - works are re-posted & remixed

**Reply by cesarenaldi** [reply]
Interesting! Given that a single Lens Publication metadata can hold references to multiple audio files via the `media` array... I can see how some of these fields could be relevant as part of an extended version of `PublicationMetadataMedia`.

Thinking out loud, should `PublicationMetadataMedia` become a union?:

```ts
type PublicationMetadataMedia = Audio | Video | Image | ...
```

so that audio specific fields can be scoped under `Audio` type. They should be more discoverable and require less documentation compared to spread them inside the existing `PublicationMetadataMedia` type.

I can also see there might be some overlap with `Video` type (see `duration`, artist`)

**Reply by cdsiren** [reply]
That approach makes sense to me!  @SweetmanTech, any thoughts?

**Reply by joshstevens19** [reply]
out of all these fields have you got 5-6 which are the most important?

**Comment by xlab** [comment]
gm! I just wanted to expand on the @haithamelmengad's high-level concern, and provide some feedback.

While designing API or a standard schema, it's important to keep abstraction at the right level so the provider-specific info doesn't creep into the container itself. While it's cool to inject multiple domain-specific fields into the general field set, we should avoid including full schemas (e.g. Catalog Metadata) because it might be inconvenient/incompatible for other external providers (e.g. SoundXYZ?).

The next question is, if we don't include full schema, at least we could include some common subset of fields, like proposed above https://github.com/lens-protocol/api-examples/discussions/101#discussioncomment-4047922 right? Well, in practice it's not always that simple to find common fields and don't miss any.

The way how it could be done in general, the metadata must have a field that keeps domain-specific metadata, that can have arbitrary schema and a protocol identifier. In this metadata, various protocols (Catalog, Sound XYZ, Stems, etc) could attach their own platform-specific schemas. Even versioned (e.g.: **Lens Metadata V3** contains a provider-specific **Catalog_V2** schema metadata field).

Various aggregators and indexers (e.g. Lenser) could then independently adopt and support particular provider's metadata and implement the mapping of relevant fields into their own internal representation.

_TL;DR: finding common fields and agreeing upon format is hard and tedious, let's abstract provider-specific details so Lens Metadata v3 is not bloated with fields that are equally suboptimal for all users on the market._

**Reply by joshstevens19** [reply]
Yes, I agree with this a lot, even though I asked for 5-6 of the most important ones. The main issue you have with this is filtering. If it's such a dynamic nature, this brings complex searching. It also hurts the type system. With graphql you rely on strongly typed properties, and in this case, it has to be fully dynamic. We have to think about this a bit more. Right now, they can do that outside of our spec, but a common object name at least gives the ability for people who use it to agree on where to fetch it from.

**Comment by GrgW** [comment]
Not thought-through so leaving here just as a FYI. 

Had a chat with qingisdead.lens about adding metadata to describe (optional) 'richer relationships' from a few weeks back: https://lenster.xyz/posts/0x0522-0x0149  _I think_ I was thinking this could help jumpstart communities within Lens. (Similar theme then to supporting Groups DMs (for Followers), but not metadata related: https://lenster.xyz/posts/0x242d-0x0138.)

At the top of that thread, you can also see qingisdead predicting a post 'type' metadata might appear for Lens: https://lenster.xyz/posts/0x0522-0x0148

**Reply by joshstevens19** [reply]
What would be in the type of post type? We have mainPublicationFocus already is that not enough?

in terms of making the describe object can you share some examples in how you would think it should look?

**Comment by EthWarrior** [comment]
For audio content this post got me thinking that having lyrics for audio metadata would be epic https://lenster.xyz/posts/0x013c1c-0xed

but alos for video could be interesting to credit authors, producers, cast etc

**Reply by joshstevens19** [reply]
I would say lyrics would just be a pointer to another file mainly because if it were dumped in the metadata itself, the storage cost for people would be huge + reading it. But do like the idea a lot. 

Credits, authors, producers, and casts are great ideas as well.

**Comment by ShravanSunder** [comment]
@joshstevens19 I think it would be good for authors to classify their posts and change their metadata after the fact. This is to make sure immutable posts can be redacted or marked falsified by the author who made the post. 

If an author marks a post this way. All the reposts and collections should reflect this meta data. I think this will help as a tool to rectify misinformation and allow authors to redact information without deleting posts. We should have a human system that forgives mistakes.

the metadata field should include date stamps and series of statuses. 

It would help with information verification proofs that balaji talked about. 

https://chainlinktoday.com/balaji-srinivasan-explains-the-pivotal-shift-from-fiat-information-to-cryptoinformation/

> information ecosystem where “anything that you’re asserting for the record” is posted on-chain via a decentralized oracle. The major advantage of this system, according to Srinivasan, is it enables the separation of facts from narrative through oracles that cryptographically check on-chain data and “advocates” that write auditable narratives based on on-chain events.

**Reply by xlab** [reply]
> If an author marks a post this way. All the reposts and collections should reflect this meta data.

Looks more like indexer-side change rather than on-chain. So maybe there is a room for standardised  indexer/ anti-spam filter metadata? Currently the spammy comments already have this flag, but looks like there could be more marks than just spam.

**Reply by ShravanSunder** [reply]
I was think if the metadata represents something from the author of the post. Especially when it comes to veracity, it should be on chain. What do you think?

**Reply by joshstevens19** [reply]
So you should post a centralised link, and we should expose a `refreshMetadata` which marks any changes to it in the DB so you can render `edited` snapshots. But yeah this is an indexer thing + feature request over metadata standard I think.

**Comment by ShravanSunder** [comment]
A field for location in metadata would help use lens for different purposes. 

- I think location can be used to create reviews for a location or business. 
- organizing posts with pictures into collages for locations 
- show what’s happening around you

**Reply by joshstevens19** [reply]
Amazing idea! We should 100% add location. Do you have any thoughts on the format of this? The geolocation API tries to set a standard

```
GeolocationPosition {coords: GeolocationCoordinates, timestamp: 1662499816712}
    coords: GeolocationCoordinates
        accuracy: 7173.528443511279
        altitude: null
        altitudeAccuracy: null
        heading: null
        latitude: 6.5568768
        longitude: 3.3488896
        speed: null
timestamp: 1662499816712
```

but `latitude and longitude` should be enough.. thoughts?

**Comment by joshstevens19** [comment]
Hey guys this is all amazing work we will be picking this up early Jan to finalise it with everyone please share as many ideas as you have before that period!

**Comment by ShravanSunder** [comment]
A thought about PublicationMainFocus. It would be good to have types for streaming (like decentralized  twitch) and audio (Twitter spaces / clubhouse)

**Reply by cesarenaldi** [reply]
Interesting, how do you see this working from a publication perspective? Is this like a channel for multiple streaming session or one of?

**Reply by joshstevens19** [reply]
This to me is a level further down like "PLATFORM": "TWITCH" for example..

**Reply by wcordelo** [reply]
gm!

Proposing an idea of adding LIVESTREAM in the PublicationMainFocus

```
export enum PublicationMainFocus {
  ...
  Livestream = 'LIVESTREAM'
}
```

@cesarenaldi @joshstevens19 @ShravanSunder I would imagine that each livestream would be its own publication. We are currently working on adding livestreaming to our platform (dragverse.io) and envision using Livepeer's streaming capabilities to work with Lens.

**Reply by joshstevens19** [reply]
I’m actually writing up a draft proposal this week and it’s already in it will share with builders this week for thoughts

**Comment by sasicodes** [comment]
gm!

Proposing an idea of adding `SHORT_VIDEO` in the **PublicationMainFocus**
```
export enum PublicationMainFocus {
  ...
  ShortVideo = 'SHORT_VIDEO'
}
```
**Difficulties**:

1. We use a separate appId (lenstube-bytes) and it works to post and retrieve short-form videos. This separate appId is an issue maintaining, analyzing, and also when calculating total interactions made in the app.
2. In case any apps want to pull only the short-form video content from the lens, they have to add respective appId (lenstube-bytes) to their system, which is not ideal for allowlist all apps in long run.
3. Vice versa, lenstube can't directly use all **video content - mainContentFocus.Video** in the portrait short-form video features like bytes.
4. There are no current possible ways to bring only short-form video content in the lens to the platform.

**Solution**:

Adding a `SHORT_VIDEO` content focus to the publication enables all types of apps to consume and post short-form videos effectively via the API.

If so, we can remove the dependency of appId for bytes and can run with a single `appId` for all the features which will be the same for all ecosystem apps planning on having short-form videos.

Hit us with feedback.
LFG 🫡

**Reply by joshstevens19** [reply]
amazing idea, what is classed as "short"? and would we expect the API to validate it within a short period? say a bad actor posts a 2 hour video with "SHORT_VIDEO"

**Reply by sasicodes** [reply]
On average, a maximum of **2mins** can be considered as SHORT, VOD services also provide mp4 streaming for videos (< 2mins) with faster load times instead of HLS streams.

In the backend, we can validate this effectively. (ofc frontends also should validate this)

**Comment by jsonpreet** [comment]
Hi Everyone,

I propose adding Media Size (for images) under ``PublicationMetadataMedia``

``interface PublicationMetadataMedia {
 ... 
  size?: string | null;
}``

**Difficulties:**

1. We are using images without their size, currently it's without width x height images because the UI messed up after image loading.

**Solution:**
Adding a ``Size`` under ``PublicationMetadataMedia`` help developers to set image sizes before loading and it's great for Nice Looking UI.

I got some feedback from users related to this issue.

Thanks!

**Reply by joshstevens19** [reply]
We are about to add a dynamic sizing ability to the publication images. This will allow you to set whatever size you wish for it to be resized. With this, do you still see the need to have the size properties on this? The main thing here is people can and will lie, and trying to do it automatically tends always to be the best bet.

**Reply by jsonpreet** [reply]
> We are about to add a dynamic sizing ability to the publication images. This will allow you to set whatever size you wish for it to be resized. With this, do you still see the need to have the size properties on this? The main thing here is people can and will lie, and trying to do it automatically tends always to be the best bet. 

It's better to add dynamic size of image in api.

**Comment by hroyo** [comment]
Hi @joshstevens19  is this appropriate thread to request customizable NFT image rendering of simple text post COLLECT?


![IMAGE 2023-03-06 11:32:39](https://user-images.githubusercontent.com/95655732/223186661-3e4513cf-6fa9-47b8-ba25-4ebeae02eb54.jpg)

**Comment by iPaulPro** [comment]
### Proposal

I'd like to propose establishing a standard for identifying the URL in a post with `LINK` set as the focus. Using a separate field has the following benefits:

- Removes ambiguity over the actual focus of a `LINK` publication.
- Allows for querying based on the URL shared
- Removes the requirement of having a (potentially long and ugly) URL in the body of the content.

The `external_url` field seems like a natural fit, but perhaps something entirely new should be used since that field is already used in various inconsistent ways.

### The problem
When creating a publication with `LINK` as the `mainContentFocus` the API doesn't currently specify how to identify the link being shared. For this reason, clients like Lenster assume the first URL in the content is the focus and automatically show rich previews of those links. This is not ideal because the post body may contain multiple links but there's no way for the user to specify which one is the primary focus.

I'm working on [Focalize](https://github.com/FocalizeApp/focalize-extension/), a browser extension for Lens that allows you to quickly share the URL of the tab you're currently on. It uses open graph meta tags to pre-populate the body. Here's an example of when it goes wrong, sharing from YouTube:

<img src="https://user-images.githubusercontent.com/354227/216040315-3c76262e-6540-44e7-bdbe-92a85f6041bd.jpg" width=320>

When viewing this post on Lenster it will appear that the focus is the shoprobuilder.com link and not the YouTube one. I can pre-populate the link at the top of the content but this is unnatural and still error prone when edits are made.

### Example use case

I have an idea for a Focalize feature where the extension automatically looks for posts shared about the webpage you're currently on. Imagine reading something on Medium and getting an unobtrusive notification telling you "This article is being discussed in 3 posts on Lens".

---
Note this was previously an issue at #121 but I've closed that and moved it here.

**Reply by cesarenaldi** [reply]
A note on the `external_url`. This pre-exist the Lens publication metadata. This is part of the original OpenSea metadata standard the Lens metadata aimed to support. From a minted collect NFT perspective this could be seen as serving a different purpose than for Lens publication content. It could be the URL where a given NFT points. It could be an external resource, it could be an accessory to the NFT, it could be integral part of a given NFT. It depends really on author's intent, difficult to predict upfront.

**Reply by iPaulPro** [reply]
> A note on the `external_url`. This pre-exist the Lens publication metadata. This is part of the original OpenSea metadata standard the Lens metadata aimed to support. From a minted collect NFT perspective this could be seen as serving a different purpose than for Lens publication content. It could be the URL where a given NFT points. It could be an external resource, it could be an accessory to the NFT, it could be integral part of a given NFT. It depends really on author's intent, difficult to predict upfront.

Yes exactly. Many Lens clients currently set the `external_url` as a link to the user's profile, for every post. This doesn't seem like a great use of the field, but as mentioned, it's already used in various inconsistent ways so creating a standard with it probably isn't ideal.

**Comment by EthWarrior** [comment]
Proposing also to add Location Anchors into the metadata. People don't necessarily have one location but they might have many locations they visit and would want to be more connected with the content or users from that area.

Location Anchors would allow to capture locations but also add multiple.

Example: London, NYC, SF, Hong Kong

Would mean that whoever is building algorithms such as content recommendation or who-to-follow algorithms, you would be able to serve more related users or content to the users.

**Comment by kkpsiren** [comment]
gm gm! 

I propose adding a new boolean parameter called `hideFromFeed`. When the default is set to `false` no breaking changes are introduced. When `hideFromFeed` is set to `true`, this hides the publication appearing in the feed. The publication will still be accessible via direct search or when accessing the main publication if it is a comment. This feature provides users with greater control over the visibility of their posts.

**Comment by mvanhalen** [comment]
How about adding an identifier for a Collection of Collects. I know normally on ETH this is based on contracts. But it can also be metadata based. We did such a thing on DeSo together with @iPaulPro. Think about adding collectionId, collectionName, collectionEdition (001 etc) collectionDescription. It could be 100% attributes (field) or become part of Lens metadata spec. Any thought about that? Or is there something else already? We would support this on [collectz.xyz](https://collectz.xyz)

**Reply by iPaulPro** [reply]
I would love to see support for collections. I believe the most important thing needed for proper collection support is using the Token ID in the metadata path to allow for ID-specific metadata. So if I mark a Collect with `isCollection` the indexer looks at `ipfs://[cid]/[token-id]` instead of just `ipfs://[cid]`.

I'm not sure how feasible this is, given the current structure of the protocol, but without id-based attributes I'm not sure how a collection would work.

**Comment by joshstevens19** [comment]
Moved to a proper LIP > https://github.com/lens-protocol/LIPs/pull/5
